src/libmw/src/node/BlockBuilder.cpp

```diff
@@ -0,0 +1,97 @@
+#include <mw/node/BlockBuilder.h>
+#include <mw/consensus/KernelSumValidator.h>
+#include <mw/consensus/Aggregation.h>
+#include <mw/consensus/Weight.h>
+
+#include <unordered_set>
+#include <numeric>
+
+MW_NAMESPACE
+
+bool BlockBuilder::AddTransaction(const Transaction::CPtr& pTransaction, const std::vector<PegInCoin>& pegins)
+{
+    // Check weight
+    uint64_t weight = Weight::Calculate(pTransaction->GetBody());
+    if ((weight + m_weight) > libmw::MAX_BLOCK_WEIGHT) {
+        LOG_ERROR("Exceeds max block weight");
+        return false;
+    }
+    
+    // Verify pegin amount matches
```

Comment:
I feel that the pegins parameter is redundant since pegins is originally formed from a call to mweb_tx.GetPegins() in mweb_miner.cpp. So verifying the pegin amount matches is guaranteed to be true.